### PR TITLE
Add typos found in Emacs and elsewhere

### PR DESCRIPTION
**Pull request by skangas**
These typos were found in Emacs as well as my dayjob.

Edited to add in alternatives as well as for consistency.

### CODE REVIEW

**Comment by skangas** [comment]
This fails on `codecov/project`, but I admit that I don't understand why exactly. This change only affects `dictionary.txt`, so it shouldn't be possible for it to reduce code coverage. Maybe I'm missing something.

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -21864,6 +21917,7 @@ eduction->education, reduction, deduction, seduction,
 eductions->reductions, deductions, educations, seductions,
 edxpected->expected
 eearly->early
+eeded->needed, heeded,
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:21920]:
```suggestion
eeded->needed, heeded, seeded, weeded,
```

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -28324,6 +28397,7 @@ gyph->glyph, gyp,
 gyphs->glyphs, gyps,
 gziniflate->gzinflate
 gziped->gzipped
+göyph->glyph
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:28400]: That one must be pretty rare :smile:

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -28776,6 +28850,7 @@ highligher->highlighter
 highlighers->highlighters
 highlighing->highlighting
 highlighs->highlights
+highlightes->highlights
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:28853]:
```suggestion
highlightes->highlights, highlighted,
```

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -32203,6 +32281,7 @@ intenationally->internationally
 intendation->indentation, intention,
 intendations->indentations, intentions,
 intendend->intended
+intendes->intends
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:32284]:
```suggestion
intendes->intends, intended,
```

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -35453,6 +35535,7 @@ medhod->method
 medhods->methods
 mediciney->medicine, medicinal,
 mediciny->medicine, medicinal,
+medicore->mediocre
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:35538]:
```suggestion
medicore->mediocre, Medicare,
```

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -46672,6 +46765,7 @@ reguster->register
 rehearsin->rehearsing, rehears in,
 rehersal->rehearsal
 rehersing->rehearsing
+rehighlightes->rehighlights
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:46768]:
```suggestion
rehighlightes->rehighlights, rehighlighted,
```

File: codespell_lib/data/dictionary.txt [hunk 7]
```diff
@@ -51041,6 +51140,7 @@ shttp->https
 shuckin->shucking, shuck in,
 shudown->shutdown
 shufle->shuffle
+shuits->shuts, suits, shits,
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:51143]:
```suggestion
shuit->shut, suit, shit,
shuits->shuts, suits, shits,
```

**Review by DimitriPapadopoulos — COMMENTED** [review]
A few suggestions. Feel free to disregard if you find my suggestions are far-stretched.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:28400]: It probably is, though it was found in Emacs. On Swedish qwerty keyboards "l" and "ö" are next to each other, so these typos are not entirely surprising.

**Comment by skangas** [comment]
> A few suggestions. Feel free to disregard if you find my suggestions are far-stretched.

Thanks, I've incorporated all your suggestions.

**Comment by DimitriPapadopoulos** [comment]
@larsoner Could you merge this one? For now, the Codecov issue is still there.

**Review by larsoner — APPROVED** [review]

**Pull request merged**